govc/sso/lpp/get.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright (c) 2022-2022 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lpp
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/govc/sso"
+	"github.com/vmware/govmomi/ssoadmin"
+	"github.com/vmware/govmomi/ssoadmin/types"
+)
+
+type get struct {
+	*flags.ClientFlag
+	*flags.OutputFlag
+}
+
+func init() {
+	cli.Register("sso.lpp.get", &get{})
+}
+
+func (cmd *get) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.ClientFlag, ctx = flags.NewClientFlag(ctx)
+	cmd.ClientFlag.Register(ctx, f)
+
+	cmd.OutputFlag, ctx = flags.NewOutputFlag(ctx)
+	cmd.OutputFlag.Register(ctx, f)
+}
+
+func (cmd *get) Description() string {
+	return `Get SSO local password policy.
+
+Examples:
+  govc sso.lpp.get
+  govc sso.lpp.get -json`
+}
+
+func (cmd *get) Process(ctx context.Context) error {
+	if err := cmd.ClientFlag.Process(ctx); err != nil {
+		return err
+	}
+	return cmd.OutputFlag.Process(ctx)
+}
+
+type lppInfo struct {
+	LocalPasswordPolicy *types.AdminPasswordPolicy
+}
+
+func (r *lppInfo) Write(w io.Writer) error {
+	fmt.Fprintf(
+		w,
+		"Description: %s\n"+
+			"MinLength: %d\n"+
+			"MaxLength: %d\n"+
+			"MinAlphabeticCount: %d\n"+
+			"MinUppercaseCount: %d\n"+
+			"MinLowercaseCount: %d\n"+
+			"MinNumericCount: %d\n"+
+			"MinSpecialCharCount: %d\n"+
+			"MaxIdenticalAdjacentCharacters: %d\n"+
+			"ProhibitedPreviousPasswordsCount: %d\n"+
+			"PasswordLifetimeDays: %d\n",
+		r.LocalPasswordPolicy.Description,
+		r.LocalPasswordPolicy.PasswordFormat.LengthRestriction.MinLength,
+		r.LocalPasswordPolicy.PasswordFormat.LengthRestriction.MaxLength,
+		r.LocalPasswordPolicy.PasswordFormat.AlphabeticRestriction.MinAlphabeticCount,
+		r.LocalPasswordPolicy.PasswordFormat.AlphabeticRestriction.MinUppercaseCount,
+		r.LocalPasswordPolicy.PasswordFormat.AlphabeticRestriction.MinLowercaseCount,
+		r.LocalPasswordPolicy.PasswordFormat.MinNumericCount,
+		r.LocalPasswordPolicy.PasswordFormat.MinSpecialCharCount,
+		r.LocalPasswordPolicy.PasswordFormat.MaxIdenticalAdjacentCharacters,
+		r.LocalPasswordPolicy.ProhibitedPreviousPasswordsCount,
+		r.LocalPasswordPolicy.PasswordLifetimeDays,
+	)
+	return nil
+}
+
+func (r *lppInfo) Dump() interface{} {
+	return struct {
```

Comment:
We did this in `sso.user.id` to combine two fields in the json output, but that breaks with the `-xml` flag. In this case, you can just do the following and `-dump` will also be cleaner:
```go
func (r *lppInfo) Dump() interface{} {
	return r.LocalPasswordPolicy
}
```